crates/ecs/src/lib.rs

```diff
@@ -68,16 +72,12 @@ pub struct Application {
 
 impl Application {
     /// Spawns a new entity in the world.
-    pub fn create_entity(&mut self) -> Entity {
-        for component_vec in self.world.component_vecs.values_mut() {
-            component_vec.push_none();
-        }
-        let entity = Entity {
-            id: self.world.entities.len(),
-            _generation: 0,
-        };
-        self.world.entities.push(entity);
-        entity
+    pub fn create_entity(&mut self) -> AppResult<Entity> {
+        let entity = self.world.create_new_entity();
+        self.world
+            .store_entity_in_archetype(entity.id, 0)
+            .map_err(ApplicationError::World)?; // todo(#72): Change so that all entities are not stored in the same big archetype, that is change index 0 to be something else.
```

Comment:
Should we propagate implementation errors to the end user? There is no way for the end user to recover from them.

Reply:
World has two types of errors: input errors and internal data structure errors. The first one can be caused by trying to borrow a component vec of a type that is not stored, these should be propagated. The other kind should just panic or we should add code for cleaning up the data structures.